### PR TITLE
Architecture explained

### PR DESCRIPTION
**Pull request by ashfame**
Markdown changes adding explanation of architecture, best viewed by browsing the markdown files - starting from [README.md](https://github.com/Automattic/neurobot/blob/architecture_explained/README.md (file in this PR)) and just click through. The new documents are [architecture.md](https://github.com/Automattic/neurobot/blob/architecture_explained/docs/architecture.md) and [toml-structure.md](https://github.com/Automattic/neurobot/blob/architecture_explained/docs/toml-structure.md).

### CODE REVIEW

**Comment by ashfame** [comment]
Will merge next week once I have had @psrpinto's feedback on the changes.

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- Query API to add more data to payload data
+- DM a certain user
+- Ask questions to a group of users in a DM and aggregate those answers & post in a matrix room. `Stand up meetings`
+- Send email
+
+## What else is planned for the future?
+
+### Keeping tabs on who's online
+
+We have a polyglots command in Automattic, which when invoked can help you find someone who speaks a certain language and is online right now. Supporting such a command would require knowing who is online and this is probably best done by `neurobot` by maintaining a "online users" list, which can simply be utilised by a workflow step.
```
Comment (psrpinto) [inline docs/architecture.md:50]: Considering this is a public repo, I'm wondering should we move this to an issue on our board instead?

**Review by psrpinto — COMMENTED** [review]
This is great! Very useful to a newcomer to the project. Thanks for taking the time to document all this.

I think we could merge it as is and it would already be great, but I think we could move some things from `architecture.md` to `README.md`, and vice-versa, which I think could make both clearer.

I noticed that currently `README.md` contains, roughly:

1. Setting up instructions
2. Architecture overview

And `architecture.md` contains:

1. Setting up instructions
    - The Components sections contains mostly information that is useful to someone who's initially setting up the project, or deploying it to production
    - The first paragraph of `How does it work` would also be something to do when initially setting up, creating the bot user, granting permissions to the bot user, etc
2. Architecture in greater detail
    - Starting from the second paragraph of `How does it work`


So my suggestions would be:

1. Move the Components section and the first paragraph of `How does it work` to the `Instructions to run it` section of `README.md`. It's possible that doing so will also require restructuring the `Instructions to run it` section but we could do that in a follow-up PR too.
2. Move the `Architecture` section from `README.md` to `architecture.md`. If you wish to keep this sections in the `README.md` because of the visuals, you could leave it as is, but I still think it would be useful to have the visual in `architecture.md` too.


I think this would result in clearer documentation because the setting-up stuff is only relevant when initially setting up the project, so I find it "muddies the waters" in `architecture.md`.

**Comment by ashfame** [comment]
Thanks! Great suggestions @psrpinto

**Pull request merged**